File: 3066. Minimum Operations to Exceed Threshold Value II/Minimum Operations to Exceed Threshold Value II.py
import heapq

nums = [2,11,10,1,3]
k = 10

sol = 0
l = []

for i in nums:
    if i<k:
        heapq.heappush(l,i)

while l:
    if len(l)==1:
        sol+=1
        break
    a, b = heapq.heappop(l), heapq.heappop(l)
    temp = a*2 + b
    if temp < k:
        heapq.heappush(l, temp)
    sol += 1
print(sol)

File: 3066. Minimum Operations to Exceed Threshold Value II/MinimumOperationstoExceedThresholdValueII.java
import java.util.PriorityQueue;

public class MinimumOperationstoExceedThresholdValueII {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        for (int i: nums){
            if (i<k) {
                heap.add(i);
            }
        }    
        int sol = 0;
        int a, b, temp;
        while (heap.size() != 0) {
            if (heap.size() == 1) {
                return sol + 1;
            }
            a = heap.poll();
            b = heap.poll();
            temp = a*2+b;
            if (temp<k && temp>=0) {
                heap.add(a*2+b);
            }
            System.out.println(a+" "+b+" "+heap.size() );
            sol++;
        }
        return sol;
    }
    public static void main(String[] args) {
        MinimumOperationstoExceedThresholdValueII mIi = new MinimumOperationstoExceedThresholdValueII();
        System.out.println(mIi.minOperations(new int[]{999999999,999999999,999999999}, 1000000000));
    }
}
